src/com/google/enterprise/adaptor/DocumentHandler.java

```diff
@@ -847,30 +859,26 @@ public OutputStream getOutputStream() throws IOException {
                   .createPipeline(os, originalContentType, metadata);
             }
           }
-        } else if (state == State.SEND_BODY_TRANSFORMED_TO_NOT_FOUND) {
-          log.log(Level.INFO, "changed SEND_BODY to NOT_FOUND {0}",
+        } else if (state == State.SEND_BODY_TRANSFORMED_TO_NO_INDEX) {
+          log.log(Level.INFO, "changed SEND_BODY to DO_NOT_INDEX {0}",
               docId.getUniqueId());
           // Not using startSending. Instead we stop header timer, start
-          // content timer, send not-found page, and setup a sink for
+          // content timer, discard metadata and content, and setup a sink for
           // bytes provided by adaptor instance itself.
           watchdog.processingCompleted(workingThread);
           watchdog.processingStarting(workingThread, contentTimeoutMillis);
-          int rc = HttpURLConnection.HTTP_NOT_FOUND;
-          HttpExchanges.startResponse(ex, rc, "text/plain", /*hasBody=*/ true);
-          countingOs = new CountingOutputStream(new CloseNotifyOutputStream(
+          // Send no body and no metadata, other than Robots noindex.
+          ex.getResponseHeaders().add("X-Robots-Tag", "noindex");
+          int rc = HttpURLConnection.HTTP_OK;
+          HttpExchanges.startResponse(ex, rc, "text/plain", /*hasBody=*/ false);
+          os = new SinkOutputStream(new CloseNotifyOutputStream(
               ex.getResponseBody()));
-          log.finest("before writing chunked not-found response");
-          countingOs.write(
-              Translation.HTTP_NOT_FOUND.toString().getBytes(ENCODING));
-          log.finest("after writing chunked not-found response");
-          os = new SinkOutputStream(countingOs);
         } else if (state == State.SEND_BODY_TRANSFORMED_TO_HEAD) {
           log.log(Level.INFO, "changed SEND_BODY to HEAD {0}",
               docId.getUniqueId());
           startSending(true);  // Lie about content. Will be 0bytes chunked.
-          countingOs = new CountingOutputStream(new CloseNotifyOutputStream(
+          os = new SinkOutputStream(new CloseNotifyOutputStream(
```

Comment:
I would appreciate anything that can be done to limit this CL's modifications because changes to this file stress my mental capabilities.  Does your comment mean that removing the COuntingOutputStream wrapper is not essential to this CL and that we can consider such removal orthogonally to this CL?  I would greatly appreciate any fragment that can be extracted and seperated be done in seperate CL. Thank you